### PR TITLE
Api to master

### PR DESCRIPTION
**Pull request by Bramgus12**
Added Swagger-UI, Routesengine and some error handling

### CODE REVIEW

File: Back-end/src/main/java/com/bramgussekloo/projects/controller/AddressController.java [hunk 1]
```diff
@@ -1,77 +1,116 @@
...
-            String output = AddressStatements.updateAddress(address);
-            if (output.equals("yes")) {
-                return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    private ResponseEntity updateAddress(
+            @ApiParam(value = "Id of the address that you want to update", required = true) @PathVariable Integer id,
+            @ApiParam(value = "The object with the address that you want to update", required = true) @RequestBody Address address
+    ) {
+        try {
+            if (id.equals(address.getId())) {
+                Address result = AddressStatements.updateAddress(address);
```
Comment (Matthbo) [inline Back-end/src/main/java/com/bramgussekloo/projects/controller/AddressController.java]: The address is updated but the server returns an error:
```
org.postgresql.util.PSQLException: No results were returned by the query.
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery(AbstractJdbc2Statement.java:255)
	at com.bramgussekloo.projects.statements.AddressStatements.updateAddress(AddressStatements.java:63)
	at com.bramgussekloo.projects.controller.AddressController.updateAddress(AddressController.java:99)
```
And
```
java.lang.IllegalArgumentException: No results were returned by the query.
	at com.bramgussekloo.projects.controller.AddressController.updateAddress(AddressController.java:106)
```

File: Back-end/src/main/java/com/bramgussekloo/projects/controller/InstituteController.java [hunk 2]
```diff
@@ -1,20 +1,145 @@
...
+            @RequestBody Institute institute
+    ) {
+        try {
+            if (id.equals(institute.getId())) {
+                return ResponseEntity.status(HttpStatus.OK).body(InstituteStatements.updateInstitute(institute));
+            } else {
+                throw new IllegalArgumentException("ID's does not match!");
+            }
+        } catch (SQLException e) {
+            throw new IllegalArgumentException(e.getMessage());
```
Comment (Matthbo) [inline Back-end/src/main/java/com/bramgussekloo/projects/controller/InstituteController.java]: Database does update but server returns error:
```
java.lang.IllegalArgumentException: No results were returned by the query.
	at com.bramgussekloo.projects.controller.InstituteController.updateInstitute(InstituteController.java:129)
```

Comment (Bramgus12) [reply Back-end/src/main/java/com/bramgussekloo/projects/controller/InstituteController.java]: Problem has been solved in the latest commit

**Comment by guanhaowu** [comment]
Werkt alleen voor de response van de server. De console in IDEAJ bevat dit nog wel.

**Pull request merged**